sdks/python/apache_beam/runners/interactive/interactive_environment.py

```diff
@@ -74,10 +83,10 @@ def __init__(self, cache_manager=None):
     self._pipeline_results = {}
     # Always watch __main__ module.
     self.watch('__main__')
-    # Do a warning level logging if current python version is below 3.5.3.
-    if sys.version_info < (3, 5, 3):
+    # Do a warning level logging if current python version is below 3.6.
+    if sys.version_info < (3, 6):
       self._is_py_version_ready = False
-      logging.warning('Interactive Beam requires Python 3.5.3+.')
+      logging.warning('Interactive Beam requires Python 3.6+.')
```

Comment:
Out of curiosity, why does it require 3.6+?

Reply:
The assertions in unittest module for if something is called such as `assert_called` are not available in some older Python versions.
To make it even worse, some features introduced in Python3 has been patched back to some Python2 versions but not some older Python3 versions. So it's possible to see tests passing in Py3.6+ and Py2.7+ but fail for Py3.5.
The 3.5.3 is only a prerequisite for dependency `jsons` (not `json`) that allows us to serialize any Python object into json format without providing schema of the data. That's how we can build dataframe from arbitrary data implicitly.

Reply:
So it sounds like the interactive functionality is available in 3.5.3, but the particular assert method we use in unit tests is only available on 3.6+, right? Would it make sense for us to support 3.5.3+, and skip those tests on <3.6? If we force 3.6+ for support, we'd lose 3.5.3+ users, right?

Perhaps it's fine to do only 3.6+, but just wanting to double click on this.